### PR TITLE
fix(editor): resolve Neovim's font like the terminal's

### PR DESCRIPTION
**Pull request by edjubert**
## Problem

Nerd Font glyphs render as tofu in the Neovim tab while the Terminal tab draws
them fine, on the same machine and the same renderer.

`NeovimPane` carried its own hardcoded font stack
(`'FiraCode Nerd Font', 'Fira Code', 'CaskaydiaCove Nerd Font', …, Menlo`,
size 13) instead of going through `useTerminalOptions` — "the one place that
decides what the terminal looks like". A user whose Nerd Font is not in that
list (Iosevka, JetBrainsMono, Hack…) falls through to `SF Mono`/`Menlo`, which
carry no Nerd Font glyphs, so the whole private-use range drops out — and the
pane also ignores both `alacritty.toml` and the Cadencr mono font setting,
including its size.

## Change

`NeovimPane` now resolves its options through `useTerminalOptions` +
`useMonoFont`, the same path as `useTerminalCoreInstanceController`: chosen
Cadencr family > `alacritty.toml` > `DEFAULT_MONO_STACK`. The block /
non-blinking cursor stays as the pane's starting shape (Neovim re-declares its
own per-mode cursor through DECSCUSR), and a terminal-config error now surfaces
in the pane instead of leaving a silent blank surface.

## Tests

`NeovimPane.test.tsx` gains four cases: the resolved font reaches the engine,
the resolution is fed by the theme palette and the chosen mono family, the
starting cursor stays a steady block, and a config error is rendered.

### CODE REVIEW

File: packages/desktop/src/components/editor/neovim/NeovimPane.tsx [hunk 1]
```diff
@@ -71,7 +77,7 @@ function NeovimPane({ featureId }: NeovimPaneProps) {
     if (status === "error") socket.detach();
   }, [status, socket.detach]);
 
-  const error = errorMessage ?? socket.lastError;
+  const error = errorMessage ?? optionsError ?? socket.lastError;
```
[the remark this answers is not in the document]
Comment (edjubert) [reply packages/desktop/src/components/editor/neovim/NeovimPane.tsx:92]: Fixed in 652746e. The pane derives its status the way `TerminalCoreInstance` already does, so an unusable config is fatal instead of pending:
```ts
const paneStatus = optionsError ? "error" : status;

useEffect(() => {
  if (paneStatus === "error") socket.detach();
}, [paneStatus, socket.detach]);
```
The socket is detached and the restart action is no longer rendered.
Recovery stays automatic: the service re-pushes the config when the file changes and `useTerminalOptions` re-resolves.

File: packages/desktop/src/components/editor/neovim/NeovimPane.tsx [hunk 2]
```diff
@@ -51,14 +46,25 @@ function NeovimPane({ featureId }: NeovimPaneProps) {
...
-      colors: theme.xterm,
-      cursor: { style: "block", blink: false },
-      scrollback: 10_000,
-    }),
-    [theme.xterm],
+  // Same resolution as the Terminal tab (`useTerminalOptions`): the chosen
+  // Cadencr mono family, else the user's alacritty.toml, else the default
+  // stack. A pane-local font list here meant Neovim rendered in whatever
+  // happened to be installed — dropping Nerd Font glyphs the terminal drew fine.
+  const { family, resolved } = useMonoFont();
```
[the remark this answers is not in the document]
Comment (edjubert) [reply packages/desktop/src/components/editor/neovim/NeovimPane.tsx:53]: I'm not taking this one.

While the setting loads, `family` is null, and that does not resolve to a hardcoded fallback.
It resolves to the alacritty.toml family:
```
family:
  appearance.fontFamily ??
  (response.found ? config.font?.normal?.family : undefined) ??
  DEFAULT_MONO_STACK,
```

A late value is applied rather than lost. `useCelerittyTerminal` pushes every later option identity into the engine, and celeritty rebuilds the glyph atlas on a font patch:

```ts
if (patch.font !== undefined) {
  this.#rebuildAtlas();
}
```
And the Terminal tab resolves the same way, with no isLoading gate:
```ts
const { family, resolved } = useMonoFont();
const monoFontFamily = family ? resolved : undefined;
```
Gating only the Neovim pane would make it diverge from the terminal, which is what this PR exists to fix, and would hold the mount behind a settings round trip.
If the transient frame is worth removing, it belongs in the mono-font layer for both surfaces.

File: packages/desktop/src/components/editor/neovim/NeovimPane.tsx [hunk 3]
```diff
@@ -67,11 +73,18 @@ function NeovimPane({ featureId }: NeovimPaneProps) {
...
+  // options that never arrive and offers a restart that only reconnects the
+  // socket. The service re-pushes the config when the file changes, so a
+  // repaired alacritty.toml recovers on its own.
+  const paneStatus = optionsError ? "error" : status;
+
   useEffect(() => {
-    if (status === "error") socket.detach();
-  }, [status, socket.detach]);
+    if (paneStatus === "error") socket.detach();
+  }, [paneStatus, socket.detach]);
```
[the remark this answers is not in the document]
Comment (edjubert) [reply packages/desktop/src/components/editor/neovim/NeovimPane.tsx]: You are right, and my previous reply was wrong on that point: `detach()` unregisters the reconnector, so nothing brought the socket back once the file was repaired.

I went the other way rather than reconnecting on a status transition. A bad config is fatal for the rendering, not for the session, so the pane no longer detaches on it:

```ts
useEffect(() => {
  if (status === "error") socket.detach();
}, [status, socket.detach]);
```

`paneStatus` still hides the restart action and shows the message. The socket is never dropped, so a repaired alacritty.toml brings the pane back with no reconnection at all. A test flips the config error away and asserts the overlay clears without a `connect()` call.

Output buffered while the renderer is unavailable stays bounded by `createTerminalTransportBridge`, which closes visibly at 1 MB instead of trimming the stream.

Pushed as 2d869b0.

File: packages/desktop/src/components/editor/neovim/NeovimPane.tsx [hunk 4]
```diff
@@ -67,11 +73,23 @@ function NeovimPane({ featureId }: NeovimPaneProps) {
...
+  const paneStatus = optionsError ? "error" : status;
+
   useEffect(() => {
+    // Fatal for the rendering, not for the session: a dead renderer ends the
+    // Neovim session, a bad config does not. `detach()` unregisters the
+    // reconnector, so detaching here would leave the pane disconnected with no
+    // automatic `connect()` once the file is repaired. Keeping the socket
+    // attached is what makes that repair recover on its own, the service
+    // re-pushes the config and `useTerminalOptions` re-resolves.
     if (status === "error") socket.detach();
```
[the remark this answers is not in the document]
Comment (edjubert) [reply packages/desktop/src/components/editor/neovim/NeovimPane.tsx]: Confirmed, and taken. `useCelerittyTerminal` keys the engine on the presence of options:

```ts
const hasOptions = options !== undefined;

useEffect(() => {
  // ...
}, [hostRef, hasOptions]);
```

so a config breaking after the pane is up disposed the engine, and nothing could repaint the replacement.

The options are now held rather than dropped. A config error is fatal only while nothing ever resolved; past that, the last good options stay in use, the engine stays alive, and a repaired file swaps the new values in through `setOptions`:

```ts
const lastGoodOptions = useRef(resolvedOptions);
if (resolvedOptions) lastGoodOptions.current = resolvedOptions;
const options = resolvedOptions ?? lastGoodOptions.current;
```

The error is still surfaced, as a toast instead of an overlay, since the session underneath is intact. Tests cover both sides: the engine keeps its options through a late config error, and the startup case stays fatal with no restart action.

This also resolves the buffering concern: with the engine never disposed, its `onData` listener is never dropped, so the bridge keeps delivering rather than accumulating toward its 1 MiB bound.

The Terminal tab disposes its engine the same way on a late config error. That is pre existing and unchanged here, and it has the shell PTY's reattachment to fall back on, so I left it out of this PR.

Pushed as 98053f7.

File: packages/desktop/src/components/editor/neovim/NeovimPane.tsx [hunk 5]
```diff
@@ -4,19 +4,64 @@ import { Loader2Icon } from "lucide-react";
...
+  const options = resolvedOptions ?? lastGoodOptions.current;
+
+  useEffect(() => {
+    if (!error || !options) return;
+    toast.error(`Terminal configuration error: ${error}`, {
+      id: `neovim-config:${featureId}`,
+    });
+  }, [error, options, featureId]);
+
+  return { options, fatalError: options ? null : error };
```
[the remark this answers is not in the document]
Comment (edjubert) [reply packages/desktop/src/components/editor/neovim/NeovimPane.tsx]: Applied in ad590d7. `.claude/rules/frontend-performance.md` does require it:

```
- **Stabilize hook return values.** A hook that returns a fresh object literal each render breaks every downstream `useMemo` and `React.memo`.
```

```ts
const fatalError = options ? null : error;
return useMemo(() => ({ options, fatalError }), [options, fatalError]);
```

Nothing downstream memoized on the object itself, the pane destructures it and only `options` reaches `useCelerittyTerminal`, so this is the convention rather than an observed regression.
